core/templates/services/exploration-html-formatter.service.ts

```diff
@@ -116,16 +115,31 @@ export class ExplorationHtmlFormatterService {
     if (labelForFocusTarget && alphabetRegex.test(labelForFocusTarget)) {
       element.setAttribute('label-for-focus-target', labelForFocusTarget);
     }
-    let propValue = parentHasLastAnswerProperty ? 'lastAnswer' : null;
+    let lastAnswerPropValue = parentHasLastAnswerProperty ? 'lastAnswer' : null;
     // The setAttribute is safe because the only possible value is 'lastAnswer'
     // as per the line above.
-    if (propValue) {
+    if (lastAnswerPropValue !== null) {
+      element.setAttribute('last-answer', lastAnswerPropValue);
       if (this.migratedInteractions.indexOf(interactionId) >= 0) {
-        element.setAttribute('[last-answer]', propValue);
-      } else {
-        element.setAttribute('last-answer', propValue);
+        element.removeAttribute('last-answer');
       }
     }
+
+    if (
+      this.migratedInteractions.indexOf(interactionId) >= 0 &&
+      (lastAnswerPropValue !== null || savedSolution === 'savedMemento()')
+    ) {
+      let interactionHtml = element.outerHTML;
+      const tagEnd = '></oppia-interactive-' + htmlInteractionId + '>';
+      let interactionHtmlWithoutEnd = interactionHtml.replace(tagEnd, '');
+      if (lastAnswerPropValue !== null) {
+        interactionHtmlWithoutEnd += `last-answer="${lastAnswerPropValue}"\n`;
```

Comment:
Shouldn't this be `interactionHtmlWithoutEnd += [last-answer]="${lastAnswerPropValue}"\n;` Same for saved solution.

Reply:
Yeah, done.